Tests/Editor/ValidatorBlacklistedClassTests.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+
+namespace DTValidator.TestBlacklisted
+{
+    public class BlacklistedOutletComponent : MonoBehaviour
+    {
+        public GameObject Outlet;
+    }
+}
+
+namespace DTValidator.Internal
+{
+    using DTValidator.TestBlacklisted;
+    using NUnit.Framework;
+
+    public static class ValidatorBlacklistedClassTests
+    {
+        private static IList<ValidatorBlacklistedClass> BlacklistedOutletComponentClassProvider()
+        {
+            var blacklistedClass = ScriptableObject.CreateInstance<ValidatorBlacklistedClass>();
+            blacklistedClass.Class = "DTValidator.TestIgnore";
+
+            return new ValidatorBlacklistedClass[] { blacklistedClass };
+        }
+
+        // [Test]
+        public static void BlacklistedMissingOutlet_ReturnsNoErrors()
+        {
+            ValidatorBlacklistedClassProvider.SetCurrentProvider(BlacklistedOutletComponentClassProvider);
+            ValidatorBlacklistedClassProvider.SetCurrentProvider(() => new ValidatorBlacklistedClass[0]);
```

Comment:
So, what the previous code was doing:
```
ValidatorIgnoredNamespaceProvider.SetCurrentProvider(IgnoredOutletComponentNamespaceProvider);
ValidatorWhitelistedNamespaceProvider.SetCurrentProvider(() => new ValidatorWhitelistedNamespace[0]);
```

was - 
line 1. overriding the IgnoredNamespaceProvider to return the hardcoded above for the test.
line 2. ensuring that there were no `ValidatorWhitelistedNamespace` objects pulled from the project directory (that might interfere with the test) by overriding it with an empty array.

Your code currently is setting the hardcoded provider correctly, but then it overrides it with an empty array. Can you instead have:
```
ValidatorBlacklistedClassProvider.SetCurrentProvider(BlacklistedOutletComponentClassProvider);
ValidatorIgnoredNamespaceProvider.SetCurrentProvider(() => new ValidatorIgnoredNamespace[0]);
ValidatorWhitelistedNamespaceProvider.SetCurrentProvider(() => new ValidatorWhitelistedNamespace[0]);
```

Reply:
Thanks. I misunderstood those lines. Tbh I didn't read precisely enough. 
So i figured this might also interfere with existing tests and changed these files as well. I hope I got all the additional lines correct in the other files.
I am just not sure about the files because I use Format on Save tool for visual studio so unfortunately it changed way more than just the additional lines and I didn't have the time to go through each changed line.